### ISSUE TITLE
Allow httpApi payload format version configuration per function

### ISSUE THREAD

**Issue by dwicker71** [serverless/serverless#9402]
### Use case description

On many of our existing AWS lambdas, we're leveraging v1.0 of the payload format version of HTTP APIs.  As we're steadily migrating lambdas to use v2.0 of the payload format version, we like to convert individual routes to v2.0 while still running others on v1.0 (all residing in the same serverless.yml file).  The provider->httpApi->payload property allows you to specify a payload version but it appears to be for all functions defined in a serverless.yml file.  It would be nice to be able to specify the payload version for individual functions. 

### Proposed solution
It seems like the better solution would be to allow each function to override the global provider->httpApi->payload setting if you wanted to run the function/lambda on a different payload format than the global setting.  This would be very helpful where you're migrating many functions from '1.0' to '2.0' payload format version and don't have to convert them together at once.

```
MyFunction:
    name: ${self:service}-MyFunction
    handler: src/handlers/MyFunction.handler
    events:
      - httpApi:
          path: /foo/bar
          method: get
          payload: '2.0'
```

### Current Workaround

We're leveraging resources.extensions to force the payload format version to '2.0' on individual routes (while leaving some using the global provider->httpApi->payload version of '1.0').

```
resources:
  extensions:
    # We're forcing the payload version to '2.0' on this route because serverless doesn't appear to honor it on the individual route settings
    # This overrides the 'PayloadFormatVersion' property on the HttpApiIntegrationMyFunction resource in the CloudFormation that's generated
    # See https://www.serverless.com/framework/docs/providers/aws/guide/resources/#override-aws-cloudformation-resource for more details
    HttpApiIntegrationMyFunction:  
      Properties:
        PayloadFormatVersion: '2.0'
```

**Comment by mark-studer** [comment]
One additional item to add is Serverless Offline already functions this way.  If you specify a different payload on the function you get a 2.0 payload but when you deploy to AWS it is configured for 1.0 and doesn't pick up the function override.

**Comment by pgrzesik** [comment]
Hello @dwicker71, thanks a lot for the proposal. I believe it makes a lot of sense to expose such option :+1: We'd be happy to accept a PR for this one :+1:

**Comment by primitivehacker** [comment]
How do we know if the version is bumped to 1 or 2 when adding the refactor? I feel like I'm missing something here or don't have enough context. Working on this while doing a live stream to help those interested in open source and any response would greatly help those who see this and are wanting to dive in to open source.

**Comment by pgrzesik** [comment]
@primitivehacker At the moment, current default is `'2.0'` and all functions use that default or an explicit setting from `provider.httpApi.payload`. Resolution of `payload` for given integration should have precedence like this: `functions[].events[].httpApi.payload` > `provider.httpApi.payload` > default

**Comment by nyamba** [comment]
Hello @pgrzesik , I have question about following case. Most functions have one route, however if they have more than one route what we should expect? following 2 routes should have same payload? or should have different payloads? Because paylodFormatVersion is only related to 'AWS::ApiGatewayV2::Integration' not routes.

```
MyFunction:  
    name: ${self:service}-MyFunction 
    handler: src/handlers/MyFunction.handler 
    events:
      - httpApi:
          path: /foo
          method: get
     - httpApi:
          path: /other
          method: get
          payload: '2.0'
```

**Comment by pgrzesik** [comment]
That is a great point @nyamba :100: In such case, we should introduce that configuration on `function` level, e.g.:

```
func:
  handler: src/handlers/MyFunction.handler 
  httpApi:
    payloadVersion: '2.0'
  events:
    - httpApi:
        path: /foo
        method: get
   - httpApi:
        path: /other
        method: get
        payload: '2.0'
```

so similarily to what we offer at `provider` level.

**Comment by medikoo** [comment]
Addressed with  #9551
